pysat/_instrument.py

```diff
@@ -1962,168 +1915,54 @@ def custom_attach(self, function, kind='modify', at_pos='end', args=[],
-                                            if len(data) > 0:
-                                                # Fixes up the incomplete check
-                                                # from above
-                                                self[vname] = data
-                                # dstr = ''.join(('Data returned will require ',
-                                #               'a "name" key in the returned ',
-                                #                 'dictionary.'))
-                                # raise DeprecationWarning(dstr)
-                            else:
-                                raise ValueError(''.join(('Must include ',
-                                                          '"name" in ',
-                                                          'returned ',
-                                                          'dictionary.')))
-                        else:
-                            raise ValueError(''.join(("kernel doesn't know ",
-                                                      "what to do with ",
-                                                      "returned data.")))
-
-                    # Modifying loaded data. Methods run on Instrument object
-                    # directly and any changes to object by the method are
-                    # retained. No data may be returned by method itself.
-                    elif kind == 'modify':
-                        null_out = func(self, *arg, **kwarg)
-                        if null_out is not None:
-                            raise ValueError(''.join(('Modify functions ',
-                                                      'should not return ',
-                                                      'any information via ',
-                                                      'return. Information ',
-                                                      'may only be propagated ',
-                                                      'back by modifying ',
-                                                      'supplied pysat object.'
-                                                      )))
-
-                    # Pass function. Method runs on a copy of Instrument,
-                    # thus no modifications to Instrument are retained,
-                    # and no data is allowed to be returned.
-                    elif kind == 'pass':
-                        tempd = self.copy()
-                        null_out = func(tempd, *arg, **kwarg)
-                        del tempd
-                        if null_out is not None:
-                            raise ValueError(''.join(('Pass functions should ',
-                                                      'not return any ',
-                                                      'information via ',
-                                                      'return.')))
+                    # Custom functions do nothing or modify loaded data. Methods
```

Comment:
My first thought for this issue was we would put a DeprecationWarning for any values of 'kind' that aren't 'modify', but leave the functionality in until the next major release. 

Taking the code out reduces our code weight which *is* nice, but given the other deprecations I'm surprised.

Reply:
I don't see any point in making the huge change of moving Custom to be a part of Instrument here and delaying the small change of removing the `kind` kwarg.  There will obviously be a deprecation warning in the last 2.X release, though.